docs/design/auditing-containment.md

```diff
@@ -0,0 +1,168 @@
+# Auditing and Node Containment
+
+## Abstract
+
+This design doc outlines how we will implement "containment mode," in which the auditing service monitors a node that has accepted an audit request, but refuses to send the requested data.
+Currently in the codebase, nodes that do this will be marked as having audit failures.
+However, we don't want to mark nodes as offline or immediately mark an audit failure if this specific case arises.
+Instead, we want to "contain" these nodes by retrying the original audit until they either eventually respond and pass the audit, or refuse to respond a certain number of times and ultimately receive an audit failure mark.
+
+## Main Objectives
+1. Nodes won’t be able to “escape” audits for specific data by refusing to send requested data.
```

Comment:
We need a clear definition of what is "refusing to send the requested data".
Consider the following use cases:
- The node is busy fulfilling other request and cannot respond to the audit request within the audit timeout defined on the satellite
- The node does not have the audit piece anymore due to SNO mistake to delete it (I did such mistake today on my storage node while cleaning up empty directories, don't ask - I am really stupid). In such the node will return an error that it does not have the piece anymore.
- The node cannot read the piece due to file permission issue - an SNO config mistake. The node will return an error.
- The node cannot read the piece due to bad sectors on the HDD. The node will return an error.

Which of the above are considered refusal to return the requested data?

Reply:
2-4 will return an error. That should still count as audit failure.

I agree. We should make it more clear.

Reply:
Yeah, I think Kaloyan had suggested we do experiments to find what errors are returned by DialNode in the different use cases, so I'll expand this section in the doc.